ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManager.java

```diff
@@ -923,10 +923,11 @@ public List<ViewManager> getOrCreateViewManagers(
       if (mViewManagers == null) {
         synchronized (mPackages) {
           if (mViewManagers == null) {
-            mViewManagers = new ArrayList<>();
+            ArrayList<ViewManager> viewManagers = new ArrayList<>();
             for (ReactPackage reactPackage : mPackages) {
-              mViewManagers.addAll(reactPackage.createViewManagers(catalystApplicationContext));
+              viewManagers.addAll(reactPackage.createViewManagers(catalystApplicationContext));
             }
+            mViewManagers = viewManagers;
```

Comment:
We probably could still get a race between this and `mViewManagers == null`, but I guess it's OK, as long as reference assignment is atomic (which it is).